### PR TITLE
[CAMEL-14581] Remove sub-menus for documentation and Reorganize Documentation menu

### PR DESCRIPTION
**Pull request by aashnajena**
- Removing sub-menus under Documentation to avoid clutter. 
- Grouping topics under documentation to improve readability

Please suggest new ways of organizing the content if necessary? I'm a little unsure about the Lists - should it be a table? Or perhaps something more creative? Please suggest!

docs_menu

### CODE REVIEW

File: content/documentations/_index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: Documentations for Apache Camel
```
Comment (zregvart) [inline content/documentations/_index.md]: I'd go with:
```suggestion
title: Apache Camel documentation
```

and place this file in `content/documentation`.

File: config.toml [hunk 2]
```diff
@@ -56,76 +56,7 @@ timeout = 300000
...
-    identifier = "camel-3-migration-guide"
-    url = "/manual/latest/camel-3-migration-guide.html"
-
-[[menu.main]]
-    name = "Camel 3.x Upgrade Guide"
-    parent = "docs"
-    weight = 10
-    identifier = "camel-3x-upgrade-guide"
-    url = "/manual/latest/camel-3x-upgrade-guide.html"
+    url = "/documentations/"
```
Comment (zregvart) [inline config.toml]: I'd go with:

```suggestion
    url = "/documentation.html"
```

File: content/documentations/_index.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## User Manual
+
+The User Manual is a comprehensive guide meant to answer all your questions - from how to begin contributing to Apache Camel, to architecture and integration patterns.
+
+ - [How to Get Started](/manual/latest/getting-started.html)
+ - [Enterprise Integration Patterns](/manual/latest/enterprise-integration-patterns.html)
+ - [Camel 2.x to 3.0 Migration Guide](/manual/latest/camel-3-migration-guide.html)
+ - [Camel 3.x Upgrade Guide](/manual/latest/camel-3x-upgrade-guide.html)
+ - [FULL USER MANUAL](/manual/latest/)
```
Comment (zregvart) [inline content/documentations/_index.md]: I'd go with:

```suggestion
 - [Complete User manual](/manual/latest/)
```

File: content/documentations/_index.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## Project Documentation
+
+Apache Camel encompasses several projects which offer many services to the user. Check out the manuals and documentation for these projects here :
+
+- [Camel-K](/camel-k/latest/)
+- [Camel Quarkus](/camel-quarkus/latest/)
+- [Camel Kafka Connector](/camel-kafka-connector/latest/)
+- [JavaDocs documentation](https://www.javadoc.io/doc/org.apache.camel/camel-api/latest/index.html) for Apache's APIs for Camel Core, Camel Catalog, Camel Cloud, Camel Cluster Support and Camel SPI etc.
```
Comment (zregvart) [inline content/documentations/_index.md]: For text I'd go with something like:

```suggestion
- [API documentation](https://www.javadoc.io/doc/org.apache.camel/camel-api/latest/index.html)
```

Though this relates more to the Camel itself not to the sub-projects, so it needs be with in the user manual section. Perhaps that section needs to be renamed also as it won't be containing just the user manual.

File: content/documentations/_index.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+title: Documentations for Apache Camel
+---
+
+## User Manual
+
+The User Manual is a comprehensive guide meant to answer all your questions - from how to begin contributing to Apache Camel, to architecture and integration patterns.
```
Comment (zregvart) [inline content/documentations/_index.md]: Perhaps something like.

```suggestion
The [User Manual](/manual/latest) is a comprehensive guide meant to help you with the key concepts of Apache Camel and software integration, from how to [begin contributing](/manual/latest/contributing.html) to Apache Camel, to [architecture](/manual/latest/architecture.html) or [integration patterns](/components/latest/eips/enterprise-integration-patterns.html).
```

File: content/documentations/_index.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+title: Documentations for Apache Camel
+---
+
```
Comment (zregvart) [inline content/documentations/_index.md]: I think we need an introduction here on how to find the right documentation and how to contribute to the documentation.

File: content/documentations/_index.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+The User Manual is a comprehensive guide meant to answer all your questions - from how to begin contributing to Apache Camel, to architecture and integration patterns.
+
+ - [How to Get Started](/manual/latest/getting-started.html)
+ - [Enterprise Integration Patterns](/manual/latest/enterprise-integration-patterns.html)
+ - [Camel 2.x to 3.0 Migration Guide](/manual/latest/camel-3-migration-guide.html)
+ - [Camel 3.x Upgrade Guide](/manual/latest/camel-3x-upgrade-guide.html)
+ - [FULL USER MANUAL](/manual/latest/)
+
+## Project Documentation
```
Comment (zregvart) [inline content/documentations/_index.md]: Perhaps:

```suggestion
## Camel sub-projects documentation
```

File: content/documentations/_index.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+Apache Camel encompasses several projects which offer many services to the user. Check out the manuals and documentation for these projects here :
```
Comment (zregvart) [inline content/documentations/_index.md]:
```suggestion
Apache Camel encompasses several sub-projects for additional platforms or ways to run Camel. Check out the individual sub-project manuals here:
```

File: content/documentations/_index.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+- [Camel-K](/camel-k/latest/)
+- [Camel Quarkus](/camel-quarkus/latest/)
+- [Camel Kafka Connector](/camel-kafka-connector/latest/)
+- [JavaDocs documentation](https://www.javadoc.io/doc/org.apache.camel/camel-api/latest/index.html) for Apache's APIs for Camel Core, Camel Catalog, Camel Cloud, Camel Cluster Support and Camel SPI etc.
+
+## Components
+
+[Component Reference Guide](/components/latest/) provides a detailed list of Camel components, their utilities and details about which versions support each component.
```
Comment (zregvart) [inline content/documentations/_index.md]: Perhaps:

```suggestion
[Component Reference](/components/latest/) is auto generated from the code to reflect the configuration options and functionality available in a particular Camel version.
```

File: layouts/documentations/documentations.html [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{{ partial "header.html" . }}
```
Comment (zregvart) [inline layouts/documentations/documentations.html]: I thought we have a default layout that looks like this one that would be used instead, less duplication that way.

Comment (aashnajena) [reply layouts/documentations/documentations.html]: How do I use the default layout? If I don't make a documentations.html file, the content doesn't seem to follow the default layout by itself.

File: config.toml (hunk 2 above)
Comment (aashnajena) [reply config.toml]: So I should not make a /documetations folder? We should just have a file?

File: content/documentations/_index.md (hunk 1 above)
Comment (aashnajena) [reply content/documentations/_index.md]: I have added all the suggestions related to rephrasing of content.

File: config.toml (hunk 2 above)
Comment (zregvart) [reply config.toml]: As we have it currently the `/documentations/` would not contain anything other than `index.html`, all the documentation we have is under `/manual/...`, `/components/...` and so on. I think it's better to have just `/documentation.html`.

File: layouts/documentations/documentations.html (hunk 10 above)
Comment (zregvart) [reply layouts/documentations/documentations.html]: Then perhaps we do not have one :(

**Comment by zregvart** [comment]
We need to rebase this, also we need to consider that the changes in #195 will conflict with the changes here so we need to work on getting one or the other merged first.

**Comment by aashnajena** [comment]
Hey, 195 should be merged first I think. As soon as that happens, I'll rebase this accordingly!

File: content/documentations/_index.md (hunk 4 above)
Comment (aashnajena) [reply content/documentations/_index.md]: For this, should I move the API documentation to User Manual (and rename it to something else)? Or let it stay under sub-projects for now?

File: layouts/documentations/documentations.html (hunk 10 above)
Comment (aashnajena) [reply layouts/documentations/documentations.html]: There does exist a _default in /layouts folder. But I don't know how to use that layout here. Can you please guide me on how to link a layout to a markdown file?

OR, should I just make a documentation.html file and write both content + layout (html) as suggested above?

File: config.toml (hunk 2 above)
Comment (aashnajena) [reply config.toml]: Shouldn't I be making a documentation.md file for content?

Comment (zregvart) [reply config.toml]: I'd put this `_index.md` in `content/docs` and use the `layout/default/section.html` layout (which should be applied by default).

File: content/documentations/_index.md (hunk 4 above)
Comment (zregvart) [reply content/documentations/_index.md]: The API documentation is for Camel not for the sub-projects, that's why I would not put the API documentation under sub-projects.

File: layouts/documentations/documentations.html (hunk 10 above)
Comment (zregvart) [reply layouts/documentations/documentations.html]: Have a look at the [template lookup order](https://gohugo.io/templates/lookup-order/) in Hugo documentation. For `_index.md` files I think `layouts/_default/section.html` is used.

File: config.toml (hunk 2 above)
Comment (aashnajena) [reply config.toml]: On it!

**Comment by aashnajena** [comment]
I think this is much cleaner now. Added `_index.md` in `/docs/` and used the `layouts/_default/section.html` template for rendering it. Thanks for the suggestions!

File: content/documentations/_index.md (hunk 6 above)
Comment (aashnajena) [reply content/documentations/_index.md]: I hope the changes are better now!

File: content/docs/_index.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+---
+title: Apache Camel Documentation
+---
+
+This page provides an overview of all kinds of documentations available for Apache Camel. Everything related to setting up, contributing, upgrading, migrating and understanding the Apache Camel architecture can be found under [User Manual](/docs/#user-manual). For documentations related to specific projects, look under the [Sub-Projects](/docs/#camel-sub-projects-documentation) section. To find component specific documentation, references are listed under [Components](/docs/#components). To contribute to the documentation, you can click on the "Edit this Page" link present at the bottom of each page.
+
+## User Manual
+
+The [User Manual](/manual/latest) is a comprehensive guide meant to help you with the key concepts of Apache Camel and software integration, from how to [begin contributing](/manual/latest/contributing.html) to Apache Camel, to [architecture](/manual/latest/architecture.html) or [integration patterns](/components/latest/eips/enterprise-integration-patterns.html).
```
Comment (zregvart) [inline content/docs/_index.md]:
```suggestion
The [User Manual](/manual/latest/) is a comprehensive guide meant to help you with the key concepts of Apache Camel and software integration, from how to [begin contributing](/manual/latest/contributing.html) to Apache Camel, to [architecture](/manual/latest/architecture.html) or [integration patterns](/components/latest/eips/enterprise-integration-patterns.html).
```

**Comment by aashnajena** [comment]
I did run yarn checks and fixed the integration patterns link, I think I forgot to commit this one. Is this ready to be merged now? Thanks!

File: content/docs/_index.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+ - [Camel 3.x Upgrade Guide](/manual/latest/camel-3x-upgrade-guide.html)
+ - [Complete User Manual](/manual/latest/)
+
+## Camel Sub-Projects Documentation
+
+Apache Camel encompasses several sub-projects for additional platforms or ways to run Camel. Check out the individual sub-project manuals here :
+
+- [Camel-K](/camel-k/latest/)
+- [Camel Quarkus](/camel-quarkus/latest/)
+- [Camel Kafka Connector](/camel-kafka-connector/latest/)
```
Comment (zregvart) [inline content/docs/_index.md]: We recently added Camel Karaf as well
```suggestion
- [Camel Kafka Connector](/camel-kafka-connector/latest/)
- [Camel Karaf](/camel-karaf-connector/latest/)
```

**Comment by aashnajena** [comment]
Added your suggestion. Does Camel Karaf need to be added under the Projects page also?

**Comment by zregvart** [comment]
> Added your suggestion. Does Camel Karaf need to be added under the Projects page also?

It's there already

**Pull request merged**